.github/workflows/codescan.yml

```diff
@@ -0,0 +1,16 @@
+name: Codescan
+on:
+  pull_request:
+    branches:
+      - main
+  push: 
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest 
+    if: ${{ !contains(github.event.head_commit.message, '[codescan skip]') }}
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v5
+    - run: pip install ruff
+    - run: ruff check .
```

Comment:
ruff check . = does this mean ruff check everything in every file, every python file, or something else?

Reply:
Yes, this will lint all Python files recursively starting at the specified directory (in this case, the root/`.` directory)